sources/engine/Xenko.Graphics/Vulkan/GraphicsDevice.Vulkan.cs

```diff
@@ -85,6 +85,27 @@ public BufferInfo(long fenceValue, SharpVulkan.Buffer buffer, DeviceMemory memor
             }
         }
 
+        private Texture _dummyTexture;
+
+        /// <summary>
+        /// A minimal 1x1 texture used while bigger textures are loading.
+        /// </summary>
+        public Texture dummyTexture
+        {
+            get
+            {
+                while( _dummyTexture.NativeImageView == ImageView.Null ) 
+                {
+                    Thread.Sleep(1); // wait until this dummy texture is ready to be used
```

Comment:
Why is this necessary?
In which case would it be accessed before being ready?
Having a while/sleep loop looks like a recipe for disaster (deadlock, etc.)

`dummyTexture` should be a simple private field and have deterministic lifetime (initialization/usage/destruction).

Reply:
It doesn't seem to be necessary on NVIDIA hardware, but it appears to be on AMD. I suspect even dummyTexture tries to get used before it is fully initialized on AMD hardware, leading to a memory access violation. This sleep holds off returning it until dummyTexture is ready, so a memory access violation should never happen on it. If you can think of a better way of handling this, please let me know... but I'm going with what works. Getting Vulkan to work on different hardware and driver implementations is dirty business, apparently.

Reply:
Not sure how this has anything to do with GPU.

The initialization is done on CPU side with code `dummyTexture = Texture.New2D(this, 1, 1, PixelFormat.R8G8B8A8_UNorm_SRgb, TextureFlags.ShaderResource);`

If nothing is using it before that, this while should never be used.
If it us used before that, it means the initialization should be done earlier.

Reply:
The crash isn't from a null dummyTexture, but a memory access violation likely from dummyTexxture.NativeImageView == null. Other textures were not having a NativeImageView set soon enough before sampling (which is why we have a dummyTexture in the first place) and it appears on AMD that dummyTexture's NativeImageView was null for too long after its CPU initialization. Hard to verify this, though, my main development machine is Intel / NVIDIA. After making this change, things worked on different AMD test machines -- so this is all the theory on why this fix worked.

Reply:
`Texture.New2D` should call `InitializeFromImpl` which should set the `NativeImageView`, so `NativeImageView` should be set as soon as texture is created.

I think this specific problem (memory access violation) is very likely due to something else (i.e. CPU/GPU sync or async resource allocation too late/freeing too early) and should be fixed very differently.

Reply:
If NativeImageView is set as soon as the texture is created, then this Thread.Sleep should never be called, and behavior wouldn't have changed due to this, and AMD hardware would still crash.

The only other change that could plausibly explain working AMD hardware is checking if VK_NV_external_memory_capabilities exists before requesting it, which is an OpenVR-required Vulkan extension. It should exist in the latest AMD drivers (although it didn't in old AMD drivers) & add it just as if it was always being added in older (and still crashing) Xenko builds...

Reply:
Hold up.... VK_NV_external_memory_capabilities

I'm guessing "NV" in that extension is referring to NVIDIA. Which means AMD wouldn't have it...

Reply:
Yes, `VK_NV_*` extensions are NVIDIA only